src/session_impl.cpp

```diff
@@ -5162,11 +5154,21 @@ namespace aux {
 		// proxy, and it's the same one as we're using for the tracker
 		// just tell the tracker the socks5 port we're listening on
 		if (m_socks_listen_socket && m_socks_listen_socket->is_open())
-			return m_socks_listen_socket->local_endpoint().port();
+		{
+			error_code ec;
+			tcp::endpoint endp = m_socks_listen_socket->local_endpoint(ec);
+			if (ec)
```

Comment:
well, it's no more dangerous than before. It was just a class of errors that wasn't handled properly (and there are still outstanding issues to be fixed there). I think a reasonable thing to do when building with no exceptions is to terminate the process any time an exception would have been thrown.

I can do that for throws from within libtorrent, but if boost::asio or new, or any other part of the standard library, throws, I can't really control that, so I'm not sure what happens. It's up to the compiler/library implementation.